tests/ui/match_single_binding.stderr

```diff
@@ -176,5 +176,25 @@ LL | |         _ => println!("Not an array index start"),
 LL | |     }
    | |_____^ help: consider using the match body instead: `println!("Not an array index start");`
 
-error: aborting due to 12 previous errors
+error: this assignment could be simplified
+  --> $DIR/match_single_binding.rs:134:5
+   |
+LL | /     val = match val.split_at(idx) {
+LL | |         (pre, suf) => {
+LL | |             println!("{}", pre);
+LL | |             suf
+LL | |         },
+LL | |     };
+   | |_____^
+   |
+help: consider removing the `match` expression
+   |
+LL ~     let (pre, suf) = val.split_at(idx);
+LL +     val = {
+LL +         println!("{}", pre);
+LL +         suf
+LL ~     };
```

Comment:
Shouldn't this suggest the following instead:

```rs
    val = {
        let (pre, suf) = val.split_at(idx);
        println!("{}", pre);
        suf
    };
```

Having the `let <pattern>` before the linted assignment can potentially shadow values. It also extends the lifetime of the pattern values. Now we can access `pre` and `suf` after the assignment. Which we couldn't previously.

Reply:
Should it? I know that's what the suggestion was in the issue, but that's how it behaved before I started this.

Reply:
Yeah, I was under the impression that that's how the suggestion behaved before. But the suggestion will introduce a change in behavior, and that's certainly not what we want.

I think @xFrednet's suggestion makes sense

Reply:
Okay, then we should change the suggestion. @Serial-ATA Could you make this change? It should be fairly simple, from what I'm guessing. If not, we can also create a new issue and merge this as it is. Your changes are already a solid improvement :upside_down_face:

Reply:
Alright, I'll try it out 🙂